app/src/main/java/org/sopt/sample/src/login/SignInActivity.kt

```diff
@@ -0,0 +1,105 @@
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.activity.result.contract.ActivityResultContracts
+import org.sopt.sample.R
+import org.sopt.sample.config.ApplicationClass
+import org.sopt.sample.config.BaseActivity
+import org.sopt.sample.databinding.ActivitySigninBinding
+import org.sopt.sample.src.HomeActivity
+
+
+class SignInActivity : BaseActivity<ActivitySigninBinding>(ActivitySigninBinding::inflate) {
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent> //회원가입 정보를 받아오기 위한 launcher
+    private var idFromSignup: String? = null
+    private var pwFromSignup: String? = null
+    private var mbtiFromSignup: String? = null
+    private lateinit var editor:SharedPreferences.Editor
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+        editor = ApplicationClass.sSharedPreferences.edit()
+
+        autoLogin()//자동 로그인
+        setResultSignUp() //회원가입 콜백
+        signIn() //로그인
+        signUp() //회원가입
+
+    }
+
+    //Signup Callback
+    private fun setResultSignUp() {
+        resultLauncher =
+            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+                if (result.resultCode == Activity.RESULT_OK) {
+                    idFromSignup = result.data?.getStringExtra("id").toString()
+                    pwFromSignup = result.data?.getStringExtra("pw").toString()
+                    mbtiFromSignup = result.data?.getStringExtra("mbti").toString()
+                    CustomSnackBar(getString(R.string.signin_success_signup)).setAnchorView(binding.signInLoginBtn)
+                        .show()
+                }
+                //회원가입 취소
+                else {
+                    CustomSnackBar(getString(R.string.signin_fail_signup)).setAnchorView(binding.signInLoginBtn)
+                        .show()
+                }
+            }
+    }
+    private fun signIn() {
+        //로그인 버튼 클릭
+        binding.signInLoginBtn.setOnClickListener {
+            //SignUp에서 받아온 id/pw와 일치하는지 체크
+            if (binding.signInIdEt.text.toString() == idFromSignup &&
+                binding.signInPwEt.text.toString() == (pwFromSignup)
+            ) {
+                showCustomToast("로그인에 성공했습니다")
+
+                //로그인 성공 시 id,pw,mbti를 sp에 저장
+                editor.putString("id",idFromSignup)
+                editor.putString("pw",pwFromSignup)
+                editor.putString("mbti",mbtiFromSignup)
+                editor.commit()
+
+                val homeIntent = Intent(this, HomeActivity::class.java)
+                homeIntent.putExtra("id", idFromSignup)
+                homeIntent.putExtra("pw", pwFromSignup)
+                homeIntent.putExtra("mbti", mbtiFromSignup)
+                startActivity(homeIntent)
```

Comment:
아래와 같이 변경해볼 수도 있겠습니다~ 개인적인 코딩 스타일이니 패스하셔도 괜찮습니다 :)
```suggestion
                Intent(this, HomeActivity::class.java).apply {
                    putExtra("id", idFromSignup)
                    putExtra("pw", pwFromSignup)
                    putExtra("mbti", mbtiFromSignup)
                }.also { intent -> // 1. "intent ->" 삭제
                    startActivity(intent) // 2. startActivity(it)도 가능. it은 Intent 타입!
                }
```
or 

```suggestion
                 val homeIntent = Intent(this, HomeActivity::class.java).apply {
                    putExtra("id", idFromSignup)
                    putExtra("pw", pwFromSignup)
                    putExtra("mbti", mbtiFromSignup)
                }
                 startActivity(homeIntent)
```